Properly building the runtimes with rpath wrappers causes the test suite to also use the wrappers.
In clang this will cause a bunch of warning due to linking flags used at compile time (-Wl,rpath=...)
that are turned into errors by `-Werror`.
`-Wno-unused-command-line-argument` also needs to be added in order to avoid the error.

Updated from 19.1.7 version.

diff --color -Naur llvm-project-23.1.0-rc3.src.orig/libcxx/utils/libcxx/test/dsl.py llvm-project-23.1.0-rc3.src/libcxx/utils/libcxx/test/dsl.py
--- llvm-project-23.1.0-rc3.src.orig/libcxx/utils/libcxx/test/dsl.py	2026-08-12 06:58:33.000000000 +0200
+++ llvm-project-23.1.0-rc3.src/libcxx/utils/libcxx/test/dsl.py	2026-08-12 15:29:00.211727605 +0200
@@ -233,7 +233,7 @@
     # fmt: off
     with _makeConfigTest(config) as test:
         out, err, exitCode, timeoutInfo, _, _ = _executeWithFakeConfig(test, [
-            "%{{cxx}} -xc++ {} -Werror -fsyntax-only %{{flags}} %{{compile_flags}} {}".format(os.devnull, flag)
+            "%{{cxx}} -xc++ {} -Werror -Wno-unused-command-line-argument -fsyntax-only %{{flags}} %{{compile_flags}} {}".format(os.devnull, flag)
         ])
         return exitCode, out, err
     # fmt: on
